openedx-unsupported/configuration
# [SE-3381] Allows adding new tinymce plugins through platform configuration

This makes it possible to install new tinymce plugins through the environment settings.

This PR is responsible for adding the tinymce plugins to the plugins directory and rebuilding the assets.

**JIRA tickets**: SE-3381, SE-3247, [OSPR-5263](https://openedx.atlassian.net/browse/OSPR-5263), [OSPR-5264](https://openedx.atlassian.net/browse/OSPR-5264)

**Dependencies**:

- https://github.com/edx/edx-platform/pull/25324

**Sandbox URL**:

- LMS: https://pr25695.sandbox.opencraft.hosting/
- Studio: https://studio.pr25695.sandbox.opencraft.hosting/

**Testing instructions**:

1. Login to Studio using `staff@example.com`/`edx`
2. Create a new Unit with an HTML Text
3. Make sure that the *ADSK Link* shows in the toolbar
4. Create a link and make sure you can preview it, and see it on the LMS

**Reviewers**

**Settings**
```yaml
EDXAPP_CMS_ENV_EXTRA:
  JS_ENV_EXTRA_CONFIG:
    TINYMCE_ADDITIONAL_PLUGINS:
      adsklink:
        linktypes:
        - download
        - offer
        filetypes:
        - PDF
        - zip
        - Video
        - Design
        orientations:
        - Vertical
        - Horizontal
        styles:
        - Primary
        - Normal
        - Secondary

TINYMCE_ADDITIONAL_PLUGINS_LIST:
- repo: https://gitlab.com/nizarmah/tinymce-adsk-plugin
  name: adsklink
  plugin_path: "/adsklink"

edx_ansible_source_repo: "https://github.com/open-craft/configuration.git"
configuration_version: "nizar/tinymce_plugins_role"
edx_platform_commit: "nizar/tinymce_plugins_combined_dependencies_master"
```

---

Make sure that the following steps are done before merging:
